forum/communications/channelsmiddleware.py

```diff
@@ -0,0 +1,46 @@
+from channels.db import database_sync_to_async
+from channels.middleware import BaseMiddleware
+from rest_framework.exceptions import APIException
+from rest_framework_simplejwt.exceptions import AuthenticationFailed
+from rest_framework_simplejwt.tokens import AccessToken
+from rest_framework_simplejwt.exceptions import TokenError, InvalidToken
+
+from users.models import CustomUser
+
+
+AUTH_HEAR_KEY = b'authorization'
+
+
+@database_sync_to_async
+def get_user(user_id: int) -> CustomUser:
+    try:
+        return CustomUser.objects.get(user_id=user_id)
+    except CustomUser.DoesNotExist:
+        raise AuthenticationFailed()
+    except Exception:
+        # TODO: call logging function
+        ...
+
+
+class JwtAuthMiddleware(BaseMiddleware):
+    def __init__(self, inner):
+        super().__init__(inner)
```

Comment:
The `__init__` method is unnecessary as it only calls the parent class's `__init__` method without any additional functionality. You may remove it unless you plan to add more functionality to it.